firmware/ventilator-controller-stm32/Core/Test/Pufferfish/Util/Ranges.txt

```diff
@@ -0,0 +1,112 @@
+      THEN('The Within function returns true') 
+
+    WHEN('The clamp function is called with the function inputs') 
+      THEN('The clamp function returns the test value (8)') 
+  
+    // value > ceiling > floor
+  GIVEN('A floor of 12 , a ceiling of 120 and test value of 200 as the function inputs') 
+    WHEN('The within function is called with the function inputs') 
+      THEN('The Within function returns false') 
+
+    WHEN('The clamp function is called with the function inputs') 
+      THEN('The clamp function returns the ceiling (120)') 
+
+Scenario:The Util::Ranges within and clamp function gives the expected result for floor = ceiling  
+    // value < floor == ceiling 
+  GIVEN('A floor of 120 , a ceiling of 120 and test value of 20 as the function inputs') 
+    WHEN('The within function is called with the function inputs ') 
+      THEN('The Within function returns false') 
+
+    WHEN('The clamp function is called with the function inputs ') 
+      THEN('The clamp function returns the floor (120)') 
+
+    // floor == value == ceiling
+  GIVEN('A floor of 20 , a ceiling of 20 and test value of 20 as the function inputs') 
+    WHEN('The within function is called with the function inputs') 
+      THEN('The Within function returns true') 
+
+    WHEN('The clamp function is called with the function inputs') 
+      THEN('The clamp function returns the floor (20)') 
+
+    // value > floor == ceiling
+  GIVEN('A floor of 10 , a ceiling of 10 and test value of 20 as the function inputs') 
+    WHEN('The within function is called with the function inputs') 
+      THEN('The Within function returns false') 
+
+    WHEN('The clamp function is called with the function inputs') 
+      THEN('The clamp function returns the floor (10)') 
+
+Scenario:The Util::Ranges within and clamp function gives the expected result for floor > ceiling
+// floor > ceiling > value
+  GIVEN('A floor of 15 , a ceiling of 8 and test value of 4 as the function inputs') 
+    WHEN('The within function is called with the function inputs') 
+      THEN('The Within function returns false') 
+
+    WHEN('The clamp function is called with the function inputs') 
+      THEN('The clamp function returns the floor (8)')
+
+    // ceiling < floor == value
+  GIVEN('A floor of 15 , a ceiling of 8 and test value of 15 as the function inputs') 
+    WHEN('The within function is called with the function inputs ') 
+      THEN('The Within function returns false') 
+
+    WHEN('The clamp function is called with the function inputs ') 
+      THEN('The clamp function returns the floor (15)') 
+    
+
+    // floor > ceiling == value
+  GIVEN('A floor of 15 , a ceiling of 8 and test value of 8 as the function inputs') 
+    WHEN('The within function is called with the function inputs ') 
+      THEN('The Within function returns true') 
+
+    WHEN('The clamp function is called with the function inputs ') 
+      THEN('The clamp function returns the floor (15)') 
```

Comment:
[proofreading] shouldn't this return the ceiling? Anyways, in your cpp file you used a test value of 15, which is inconsistent with this txt file. Also, your GIVENs/WHENs/THENs in the cpp file don't match the ones in this txt file. When I review tests I expect to be able to assume that the GIVENs/WHENs/THENs match between the files.